doc_source/add-aws-mobile-nosql-database.rst

```diff
@@ -140,13 +139,78 @@ Connect to your backend
                      Thread mythread = new Thread(runnable);
                      mythread.start();
 
+   Android - Kotlin
+      #. Set up AWS Mobile SDK components with the following
+         :ref:`add-aws-mobile-sdk-basic-setup` steps.
+
+         #. :file:`app/build.gradle` must contain:
+
+            .. code-block:: java
+               :emphasize-lines: 2
+
+                dependencies{
+                    implementation 'com.amazonaws:aws-android-sdk-ddb-mapper:2.6.+'
+                }
+
+         #. For each Activity where you make calls to perform database operations, import the
+            following APIs.
+
+            .. code-block:: java
+               :emphasize-lines: 1
+
+                import com.amazonaws.mobileconnectors.dynamodbv2.dynamodbmapper.DynamoDBMapper;
+
+      #. Create a :code:`DynamoDBMapper` client for your app as in the following
+         example.
+
+         .. code-block:: kotlin
+            :emphasize-lines: 2, 9-13
+
+             // import DynamoDBMapper
+             import com.amazonaws.mobileconnectors.dynamodbv2.dynamodbmapper.DynamoDBMapper;
+
+             class MainActivity : AppCompatActivity() {
+               private var dynamoDBMapper: DynamoDBMapper? = null
+
+               override fun onCreate(savedInstanceState: Bundle?) {
+                 super.onCreate(savedInstanceState)
+                 setContentView(R.layout.activity_main)
+
+                 val client = AmazonDynamoDBClient(AWSMobileClient.getInstance().credentialsProvider)
+                 dynamoDBMapper = DynamoDBMapper.builder()
+                    .dynamoDBClient(client)
+                    .awsConfiguration(AWSMobileClient.getInstance().configuration)
+                    .build()
+               }
+            }
 
+      #. Add the project data model files you downloaded from the
+         |AMH| console. The data models provide set and get methods for each attribute of a |DDB|
+         table they model.
+
+         #. Copy the data model file(s) you downloaded,
+            :file:`./YOUR-PROJECT-NAME-integration-lib-aws-my-sample-app-android/src/main/java/com/amazonaws/models/nosqlYOUR-TABLE-NAMEDO.java` into the Android Studio folder that contains your main activity.
+
+
+      .. list-table::
+         :widths: 1
+
+         * - .. note:: **Use Asynchronous Calls to DynamoDB**
```

Comment:
Dynamo has an async client. Should we recommend that instead?

Reply:
Noooo!!!!!  You want to use an ASyncTask instead so that you can get the background thread to UI thread translation done for you.